content/5_Plat/Bin_Jump.mdx

```diff
@@ -215,9 +214,100 @@ export const problems = {
 	<Resource source="SecondThread" title="Tree Basics - Binary Lifting" url="https://codeforces.com/blog/entry/81527"> </Resource>
 </Resources>
 
+<Info>
+
+Binary jumping is more commonly referred to as "binary lifting."
+
+</Info>
+
 ### Solution
 
-<IncompleteSection />
+To solve this problem, we can use a standard binary lifting implementation where ```query(int v, int j)``` corresponds to the $j$th ancestor of v.
+
+In our ```query(int v, int j)``` if we ever encounter a value of $-1$ or $0$, then such a node does not exist and we can simply return its value. This is because the lowest number a node can be is $1$ (the root of the tree).
+
+To calculate the columns required for the ```up``` array, use the formula $\textrm{ceil}(\log_2{N})+1$ which in our case simplifies to $\textrm{ceil}(\log_2{(2\cdot 10^5)}) + 1 = 19$.
+
+<Info>
+
+It never hurts to add additional columns (as long as it's reasonable)!
+
+</Info>
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define FORE(i, a, b) for(int i = (a); i <= (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+
+int N, Q;
+
+const int MX = 2e5+5;
+
+int up[MX][20]; // 19 is calculated with our expression but we can add padding to be safe.
+
+vector<int> adj[MX]; // adjacency list
+
+void dfs(int v, int p) { // we keep track of the current node v and its parent p
+    up[v][0] = p;
+
+    FOR(i, 1, 20) {
```

Comment:
probably more convenient to let the root to be 0 and let it point to itself (`up[0][0]=0`) like here:

https://github.com/bqi343/USACO/blob/master/Implementations/content/graphs%20(12)/Trees%20(10)/LCAjump.h

No need to introduce `-1`.